internal/controller/accessrequest_controller.go

```diff
@@ -275,6 +323,120 @@ func (r *AccessRequestReconciler) getRoleTemplate(ctx context.Context, ar *api.A
 	return roleTemplate, nil
 }
 
+// handleTTL checks whether the AccessRequest has exceeded its configured TTL (Time-To-Live) duration.
+// If the TTL is not configured, it returns false and does not mark the resource for deletion.
+// If the TTL is exceeded, it marks the resource for deletion by adding a deletion timestamp.
+// TTL is only applied for concluded AccessRequests (i.e., those in Denied, Expired, etc. See isConcluded()).
+//
+// Parameters:
+// - ctx: The context for the operation, used for cancellation and deadlines.
+// - ar: The AccessRequest object to check and potentially mark for deletion.
+//
+// Returns:
+// - A boolean indicating whether the TTL was exceeded.
+// - An error if adding the deletion timestamp fails, or nil if successful.
+func (r *AccessRequestReconciler) handleTTL(ctx context.Context, ar *api.AccessRequest) (bool, error) {
+	// Skip if the TTL is not configured.
+	if !hasTTLConfig(r.Config) {
+		return false, nil
+	}
+	// Skip if the AccessRequest is not concluded.
+	if !isConcluded(ar) {
+		return false, nil
+	}
+
+	// Check if the AccessRequest has exceeded its configured TTL (Time-To-Live) duration.
+	ttlExceeded := time.Now().After(getTTLTime(ar, r.Config))
+
+	if ttlExceeded {
+		// If TTL is exceeded, set the resource to be deleted.
+		err := r.Delete(ctx, ar)
+		if err != nil {
+			return ttlExceeded, fmt.Errorf("error adding deletion timestamp: %w", err)
+		}
+	}
+	return ttlExceeded, nil
+}
+
+// getTTLTime calculates the expiration time for an AccessRequest.
+// It adds the TTL (Time-To-Live) duration, defined in the controller configuration,
+// to the creation timestamp of the AccessRequest.
+//
+// Parameters:
+// - ar: The AccessRequest object containing the creation timestamp.
+// - config: The controller configuration providing the TTL duration.
+//
+// Returns:
+// - The calculated expiration time as a time.Time object.
+func getTTLTime(ar *api.AccessRequest, config config.ControllerConfigurer) time.Time {
+	return ar.GetCreationTimestamp().Time.Add(config.ControllerAccessRequestTTL())
```

Comment:
It seems weird to have the TTL added to the creation time, especially since this TTL is only valid for a concluded access request.

Instead, I think the TTL should be added to the time at which the resource transitioned to a "concluded" state.

So if my TTL is 15m, then I want my access request to exist for an additional 15 minutes after it has concluded. Otherwise, the TTL directly depends on how much time the access is valid for, and how fast the ephemeral system is.